File: examples/Go/mdclient.go
//  Majordomo Protocol client example
//  Uses the mdcli API to hide all MDP aspects
//  
//  To run this example, you may need to run multiple *.go files as below
//  go run mdp.go zhelpers.go mdcliapi.go mdclient.go [-v]
//
//  Author: iano <scaly.iano@gmail.com>

package main

import (
	"fmt"
	"os"
)

func main() {
	verbose := len(os.Args) >= 2 && os.Args[1] == "-v"

	client := NewClient("tcp://localhost:5555", verbose)
	defer client.Close()

	count := 0
	for ; count < 1e5; count++ {
		request := [][]byte{[]byte("Hello world")}
		reply := client.Send([]byte("echo"), request)
		if len(reply) == 0 {
			break
		}
	}

	fmt.Printf("%d requests/replies processed\n", count)
}

File: examples/Haskell/mdclient.hs
import MDClientAPI

import ZHelpers

import System.Environment
import Control.Monad (forM_, when)
import Data.ByteString.Char8 (unpack, pack)

main :: IO ()
main = do
    args <- getArgs
    when (length args /= 1) $
        error "usage: mdclient <is_verbose(True|False)>"
    let isVerbose = read (args !! 0) :: Bool

    withMDCli "tcp://localhost:5555" isVerbose $ \api ->
        forM_ [0..10000] $ \i -> do
            mdSend api "echo" [pack "Hello world"]
